Spring Rest Docs 적용

dnjwm8612 opened the pull request:
- Spring Rest Docs 적용
- 투표 관련 단위 테스트 추가
- 회원 관련 단위 테스트 보완
- adoc 파일 작성
- adoc 파일 활용한 API 문서 자동화 출력(/build/asciidoc/html5/api-doc.html)

dnjwm8612 committed:
Add Spring_Rest_Docs 의존성과 플러그인 추가 및 설정 (#38)

dnjwm8612 committed:
Add Spring_Rest_Docs 의존성과 플러그인 추가 및 설정

dnjwm8612 committed:
Merge remote-tracking branch 'origin/feature/38' into feature/38

dnjwm8612 committed:
Merge branch 'develop' into feature/38

dnjwm8612 committed:
Add 투표 생성, 목록 조회, 상세 조회 단위 테스트 추가 (#38)

dnjwm8612 committed:
Add 투표 제거 단위 테스트 추가 및 투표 클래스 리팩토링 (#38)
- 정상적으로 투표 제거 기능을 작동하는 시나리오 테스트 작성
- 테스트 진행 시 테스트 객체를 생성을 위해 생성자를 추가하는 방식을 사용할 경우 테스트 상황에 따라 생성자가 추가되고 그로 인해 가독성이 떨어지므로 빌드 방식으로 변경
- 빌드 방식으로 변경에 따른 테스트 클래스 생성자 부분 코드 리팩토링 진행

dnjwm8612 committed:
Fix 투표 테스트 클래스 리팩토링 (#38)
- 테스트 직관성을 위해 테스트 대상 메소드_상황_결과 형식으로 테스트 메소드 이름 변경
- 목록 조회, 상세 조회, 생성, 삭제 기능 잘못된 요청이 들어올 경우 테스트 확인 메소드 추가

dnjwm8612 committed:
Add 투표 기능 테스트 클래스 Spring Rest Docs 적용 (#38)

dnjwm8612 committed:
Fix 투표 기능 테스트 클래스 보완 (#38)
- 투표 생성 기능 테스트 진행 결과 투표 클래스에 votingWriter @NotEmpty 적용에 대해서 문제 발생으로 인해 @NotNull로 변경
- 투표 테스트 클래스 투표 생성 기능 테스트 부분 content 부분 {} 추가

dnjwm8612 committed:
Add 투표수 증가 단위 테스트 추가 및 Spring Rest Docs 적용 (#38)
- 투표 아이템 테스트 클래스 추가 후 투표수 증가 기능 단위 테스트 추가
- 추가한 단위 테스트 Spring Rest Docs 적용

dnjwm8612 committed:
Fix 투표 기능 테스트 클래스 네이밍 규칙 안 맞는 메소드 수정 (#38)
- 정상적인 투표 기능 테스트 valid 부분 Valid로 변경

dnjwm8612 committed:
Add 투표 서비스 테스트 클래스와 기능 단위 테스트 추가 (#38)

dnjwm8612 committed:
Add 투표 서비스 테스트 클래스와 기능 단위 테스트 추가

dnjwm8612 committed:
Fix 투표 컨트롤러 테스트 클래스 정상 동작 단위 테스트 이름 변경 (#38)
- 정상 동작 단위 테스트 대소문자 규칙 안 맞는 부분 수정 (true -> True)

dnjwm8612 committed:
Fix 투표 컨트롤러 테스트 클래스 정상 동작 단위 테스트 이름 변경
- 정상 동작 단위 테스트 대소문자 규칙 안 맞는 부분 수정 (true -> True)

dnjwm8612 committed:
Merge remote-tracking branch 'origin/feature/38' into feature/38

dnjwm8612 committed:
Add 투표 아이템 서비스 테스트 클래스와 기능 단위 테스트 추가 (#38)

dnjwm8612 committed:
Fix 투표 서비스 테스트 클래스 기능 테스트 보완 (#38)
- 일정하지 않은 테스트 메소드 수정
- 생성 요청 처리 단위 테스트 부분 정상적인 동작 테스트를 위해 테스트 객체 추가 및 정상 동작 확인
- 단위 테스트 설명 추가

dnjwm8612 committed:
Add 투표 저장 기능 예외 처리 핸들러 추가 (#38)

dnjwm8612 committed:
Add 투표 서비스 테스트 클래스 예외 관련 단위 테스트 추가 (#38)
- 목록 조회, 생성, 삭제 처리 과정에서 발생하는 예외 처리 과정에 대한 단위 테스트 추가

dnjwm8612 committed:
Add 테스트 코드에 Rest Docs 적용 (#38)
- 기존에 작성된 컨트롤러 테스트 클래스 단위 테스트에 Rest Docs 적용
 - Rest Docs 적용한 테스트 실행하여 정상적으로 테스트 통과 시 snippets 정상적으로 생성 확인

dnjwm8612 committed:
Fix 투표 컨트롤러, 서비스 클래스 상세 조회 메소드 parameter 수정 (#38)
- Rest Docs 적용 과정에서 투표 상세 조회 투표 컨트롤러 및 서비스 클래스에서 parameter를 직관적인 단어로 변경

dnjwm8612 committed:
Add snippets를 활용하여 투표 API 문서 작성 (#38)

dnjwm8612 committed:
Fix 투표 기능 관련 클래스 보완 작업 진행 (#38)
- 투표 아이템 서비스 클래스 설명 추가 작업
- 테스트 클래스 설명 추가 및 잘못된 설명 수정 작업
- 투표 서비스 테스트 클래스 목록 조회 동작 확인 부분 추가
- 투표 아이템 서비스 테스트 클래스 VotingItemMapper 초기화 설정 부분 불필요하다 판단하여 제거 후 필요한 부분만 Method로 이동

dnjwm8612 committed:
Fix Curator 테스트 클래스 보완 작업 (#38)
- Rest Docs 작성 완료 빌드 진행 시 Curator 테스트 클래스 오류 인해 빌드 불가로 인해 보완 작업 진행

dnjwm8612 committed:
Fix 투표 기능 테스트 클래스들 CheckStyle 보완 (#38)

phantom08266 commented:
Can one of the admins verify this patch?

phantom08266 commented:
@dnjwm8612  병준님 해당 코드에서 resultType이 저희 코드기준 model에 있는 클래스들이어야 정상적으로 파싱이 되는 걸로 알고있습니다. 해당 부분으로인해 CuratorServieTest 등 일부 테스트가 통과되지 않았던 것 같습니다. 아마도 이것때문에 기존에 작성했던 통합테스트를 단위테스트로 전부 변경하신것 같아 보입니다. 

https://github.com/f-lab-edu/careers/blob/f88c4cdb717799635ec787515a70991b9a93b004/src/main/resources/mybatis/VotingMapper.xml#L23-L26

dnjwm8612 commented:
> @dnjwm8612 병준님 해당 코드에서 resultType이 저희 코드기준 model에 있는 클래스들이어야 정상적으로 파싱이 되는 걸로 알고있습니다. 해당 부분으로인해 CuratorServieTest 등 일부 테스트가 통과되지 않았던 것 같습니다. 아마도 이것때문에 기존에 작성했던 통합테스트를 단위테스트로 전부 변경하신것 같아 보입니다.
> 
> https://github.com/f-lab-edu/careers/blob/f88c4cdb717799635ec787515a70991b9a93b004/src/main/resources/mybatis/VotingMapper.xml#L23-L26

아. 죄송합니다.
우선 제가 작성된 부분을 제대로 확인을 안하고 빌드 오류 문구만 보고 말씀드린 것 같습니다.
수정과 테스트 다시 진행해 보겠습니다.

dnjwm8612 committed:
Revert "Fix Curator 테스트 클래스 보완 작업 (#38) - Rest Docs 작성 완료 빌드 진행 시 Curator 테스트 클래스 오류 인해 빌드 불가로 인해 보완 작업 진행"

This reverts commit f88c4cdb

dnjwm8612 committed:
Fix 투표 조회 기능 Mppaer 수정 (#38)
- 투표 목록 조회 기능 쿼리와 반환 요소 일치하지 않는 부분 수정
- 투표 상세 조회 시 투표 아이템과 해당 투표 아이디도 같이 반환하도록 변경

dnjwm8612 committed:
Fix 투표 삭제 동작 순서 변경 (#38)
- 투표 아이템 투표 테이블에 투표 아이디는 투표 테이블에 아이디를 포린키를 받고 사용하고 있기 때문에 투표 아이템 삭제 후 투표 삭제 진행 순서로 변경

dnjwm8612 committed:
Fix 투표 목록 조회 부분 문제 수정 (#38)
- 투표 서비스 클래스에 투표 목록 조회에서 사용하는 int 타입인 limit 변수를 생성자 bean 주입하려는 문제 발생하여 final 제거 AllArgsConstructor에서 RequiredArgsConstructor으로 변경
- 투표 Mapper resultMap 중복 문제 발생으로 해결하기 위해 resultMap 제거

dnjwm8612 commented:
> @dnjwm8612 병준님 해당 코드에서 resultType이 저희 코드기준 model에 있는 클래스들이어야 정상적으로 파싱이 되는 걸로 알고있습니다. 해당 부분으로인해 CuratorServieTest 등 일부 테스트가 통과되지 않았던 것 같습니다. 아마도 이것때문에 기존에 작성했던 통합테스트를 단위테스트로 전부 변경하신것 같아 보입니다.
> 
> https://github.com/f-lab-edu/careers/blob/f88c4cdb717799635ec787515a70991b9a93b004/src/main/resources/mybatis/VotingMapper.xml#L23-L26

Curator 테스트 부분 revert 진행 후 해당 및 관련 수정 진행하였습니다.

f-lab-dev commented:
CI가 깨지고있는데 확인 한번 부탁드립니다~

dnjwm8612 commented:
> CI가 깨지고있는데 확인 한번 부탁드립니다~

현재 CI 깨지는 부분에 대해서는 계속 조처를 하고 있지만 정확한 대처 방법을 찾지 못하고 있습니다.

f-lab-dev commented:
로그를 자세히 살펴봐야겠네요~

dnjwm8612 committed:
Fix Spring Rest Docs CI 적용 (#38)
- Spring Rest Docs 적용을 위해 단위 테스트를 진행이 필수적이기 때문에 단위 테스트 생략 부분 제거

dnjwm8612 committed:
Fix Spring Rest Docs 관련 CI 문제 수정 (#38)

dnjwm8612 committed:
Fix CD 변경 (#38)
- Spring Rest Docs 적용하여 정상적인 빌드 진행을 위해 변경
